### DISCUSSION TITLE
RClone instead of raw FUSE: Could We / Should We ?

### DISCUSSION THREAD

**Discussion by dalloriam** [menmos/menmos#71] [Ideas]
Right now [rclone](https://rclone.org/) supports mounting virtual filesystems from various backends. Their code is [very robust](https://rclone.org/commands/rclone_mount/) and already covers most/all of our nasty edgecases (caching, disconnections, etc.)

We could fork/adapt RClone's codebase (as it is MIT licensed) to only have a menmos backend, and let rclone take care of caching & syncing and the likes.

Thoughts? @laurentlp @gabrielmethot

**Comment by dalloriam** [comment]
The main upsides:
- Cross-platform support for free
- Battle-tested abstractions + cache implementations

The main downsides I see:
- Do we want to re-write & maintain a menmos client in go?
- What if rclone does something in a way that makes the mounts _much_ slower for our use case? Do we start adapting their code to fit our use-case?

Do you think we can figure this out _without_ trying it? And if not, should we try it?

**Reply by gabrielmethot** [reply]
You should try to make a MVP using their code to determine whether or not every important requirement is satisfied or not.

> Do we want to re-write & maintain a menmos client in go?

> What if rclone does something in a way that makes the mounts much slower for our use case? Do we start adapting their code to fit our use-case?

I wouldn't worry about it for now. Let's first figure out if it works or not, regardless of the performance/maintenance issues.
